data_transfer/lib/byteflies.py

```diff
@@ -96,32 +100,38 @@ def get(self, signal_id: str, algorithm_id: str = "") -> Any:
 
 
 def download_file(
-    download_folder: Path,
     session: requests.Session,
+    download_folder: Path,
     studysite_id: str,
     recording_id: str,
     signal_id: str,
     algorithm_id: str = "",
-) -> bool:
+) -> Union[bool, str]:
     """
     Download all files associated with one ByteFlies recording.
     """
-    details: dict = __get_recording_by_id(session, studysite_id, recording_id)
-    signal: dict = next((s for s in details["signals"] if s["id"] == signal_id), None)
-    url, filename = (
-        (signal["rawData"], signal_id)
-        if not algorithm_id
-        else (
-            __get_algorithm_uri_by_id(
-                session, studysite_id, recording_id, algorithm_id
-            ),
-            algorithm_id,
+    try:
+        details: dict = __get_recording_by_id(session, studysite_id, recording_id)
+        signal: dict = next(
+            (s for s in details["signals"] if s["id"] == signal_id), None
+        )
+        url, filename = (
+            (signal["rawData"], signal_id)
+            if not algorithm_id
+            else (
+                __get_algorithm_uri_by_id(
+                    session, studysite_id, recording_id, algorithm_id
+                ),
+                algorithm_id,
+            )
         )
-    )
 
-    # TODO: for now, assumes that this method never throws ...
-    __download_file(download_folder, url, filename)
-    return True
+        if __download_file(download_folder, url, filename):
+            return filename
+        return False
+    except requests.HTTPError:
+        log.error(f"GET Exception to {url} ", exc_info=True)
+        return False
```

Comment:
I would return `None` in place of `False` given it would have the same outcome in an IF statement and is more descriptive of intent, i.e., if there is no filename then the string is non-existing. We could use "" instead, but imho None is more explict.